test/core/OriginCore.js

```diff
@@ -0,0 +1,263 @@
+            await originCore.reportBlock(
+                expectedBlockHashOne,
+                blockHeight,
+                0,
+                '0x82ba3527d2433d8a6c50502f341f165955585b1337512e5cd9550eef623c8fd0',
+                '0x82ba3527d2433d8a6c50502f341f165955585b1337512e5cd9550eef623c8fd0'
+            );
+            await originCore.reportBlock(
+                expectedBlockHashTwo,
+                blockHeight,
+                0,
+                '0x12345627d2433d8a6c50502f341f165955585b1337512e5cd9550eef62312345',
+                '0x12345627d2433d8a6c50502f341f165955585b1337512e5cd9550eef62312345'
+            );
+
+            let reportedHashes = await originCore.getReportedBlockHashes.call(blockHeight);
+            // There were two blocks reported in order.
+            assert.strictEqual(
+                reportedHashes.length,
+                2,
+                'Reporting two blocks should lead to two stored hashes.'
+            );
+            assert.strictEqual(
+                reportedHashes[0],
+                expectedBlockHashOne,
+                'The first hash of the reported block was not stored correctly.'
+            );
+            assert.equal(
+                reportedHashes[1],
+                expectedBlockHashTwo,
+                'The second hash of the reported block was not stored correctly.'
+            );
+
+        });
+
+        it('should reject a report at the wrong height', async () => {
+            let expectedBlockHash = '0xb59b762b2a1d476556dd6163bc8ec39967c4debec82ee534c0aed7a143939ed2';
+            let blockHeight = 17;
+
+            await ost.approve(originCore.address, 10 ** 18);
+
+            let hadError = false;
+            try {
+                await originCore.reportBlock(
+                    expectedBlockHash,
+                    blockHeight,
+                    0,
+                    '0x82ba3527d2433d8a6c50502f341f165955585b1337512e5cd9550eef623c8fd0',
+                    '0x82ba3527d2433d8a6c50502f341f165955585b1337512e5cd9550eef623c8fd0'
+                );
+            } catch (error) {
+                // TODO: Truffle v5 will support require messages with web3 1.0
+                assert(
+                    error.message.search('revert') > -1,
+                    'The contract should revert. Instead: ' + error.message
+                );
+                hadError = true;
+            }
+
+            assert(
+                hadError,
+                'The contract should throw an error when the height is wrong.'
+            );
+        });
+
+        it('should reject a report with the wrong hash', async () => {
+            let invalidBlockHash = '0x0000000000000000000000000000000000000000000000000000000000000000';
+            let blockHeight = 0;
+
+            await ost.approve(originCore.address, 10 ** 18);
+
+            let hadError = false;
+            try {
+                await originCore.reportBlock(
```

Comment:
can we not use the `ExpectThrow`?

Reply:
We cannot as it also succeeds on "out of gas" and "invalid opcode". Instead I will add an `expectRevert`.